File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length;
        if (n <= 2) {
            return n;
        }

        // Find min and max values
        int a = nums[0];
        int b = nums[0];
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] < a) {
                a = nums[i];
            }
            if (nums[i] > b) {
                b = nums[i];
            }
        }

        int l = 0;
        int r = n - 1;
        int i = 0;
        int j = 0;
        int p = 0;
        int q = 0;

        // 1. First target from front
        while (nums[l] != a && nums[l] != b) {
            i++;
            l++;
        }
        i++;

        // 2. First target from back
        while (nums[r] != a && nums[r] != b) {
            j++;
            r--;
        }
        j++;

        // 3. Both from front
        int k = 0;
        int s = 2;
        while (s > 0) {
            if (nums[k] == a || nums[k] == b) {
                s--;
            }
            p++;
            k++;
        }

        // 4. Both from back
        k = n - 1;
        s = 2;
        while (s > 0) {
            if (nums[k] == a || nums[k] == b) {
                s--;
            }
            q++;
            k--;
        }

        return Math.min(p, Math.min(q, i + j));
    }
}

File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        if len(nums)<=2 :
            return len(nums)
        a,b=min(nums),max(nums)
        l,r=0,len(nums)-1
        i,j=0,0
        p,q=0,0
        while nums[l]!=a and nums[l]!=b :
            i+=1
            l+=1
        i+=1
        while nums[r]!=a and nums[r]!=b :
            j+=1
            r-=1
        j+=1
        k,s=0,2
        while s>0 :
            if nums[k]==a or nums[k]==b :
                s-=1
            p+=1
            k+=1
        k,s=len(nums)-1,2
        while s>0 :
            if nums[k]==a or nums[k]==b :
                s-=1
            q+=1
            k-=1
        return min(p,q,(i+j))